book/src/hints.md

```diff
@@ -1,54 +1,75 @@
 # Display hints
 
-A hint can be applied to each formatting parameter to change how it's printed on the host side.
-The hint follows the syntax `:x` and must come after the type within the braces.
-Examples: typed `{=u8:x}`, untyped `{:b}`
+> A hint can be applied to each formatting parameter to change how it's printed on the host side.
+
+The hint follows the syntax `:Display` and must come after the type within the braces, for example:
+* typed `{=u8:x}`
+* untyped `{:b}`
 
 The following display hints are currently supported:
 
-- `x`, lowercase hexadecimal
-- `X`, uppercase hexadecimal
-- `?`, `core::fmt::Debug`-like
-- `b`, binary
-- `a`, ASCII
-- `us`, microseconds (formats integers as time stamps)
+| hint  | name                                           |
+| :---- | :--------------------------------------------- |
+| `:x`  | lowercase hexadecimal                          |
+| `:X`  | uppercase hexadecimal                          |
+| `:?`  | `core::fmt::Debug`-like                        |
+| `:b`  | binary                                         |
+| `:a`  | ASCII                                          |
+| `:us` | microseconds (formats integers as time stamps) |
 
-The first 4 display hints resemble what's supported in `core::fmt`. Examples below:
+The first 4 display hints resemble what's supported in `core::fmt`, for example:
 
 ``` rust
 # extern crate defmt;
-defmt::info!("{=u8:x}", 42); // -> INFO 2a
-defmt::info!("{=u8:X}", 42); // -> INFO 2A
+defmt::info!("{=u8:x}", 42);  // -> INFO 2a
+defmt::info!("{=u8:X}", 42);  // -> INFO 2A
 defmt::info!("{=u8:#x}", 42); // -> INFO 0x2a
-defmt::info!("{=u8:b}", 42); // -> INFO 101010
+defmt::info!("{=u8:b}", 42);  // -> INFO 101010
 
 defmt::info!("{=str}", "hello\tworld");   // -> INFO hello    world
 defmt::info!("{=str:?}", "hello\tworld"); // -> INFO "hello\tworld"
 ```
 
-Leading zeros are supported, for example
+The ASCII display hint formats byte slices (and arrays) using Rust's byte string syntax.
 
 ``` rust
 # extern crate defmt;
-defmt::info!("{=u8:03}", 42); // -> INFO 042
-defmt::info!("{=u8:08X}", 42); // -> INFO 0000002A
-defmt::info!("{=u8:#010X}", 42); // -> INFO 0x0000002A
+let bytes = [104, 101, 255, 108, 108, 111];
+
+defmt::info!("{=[u8]:a}", bytes); // -> INFO b"he\xffllo"
 ```
 
-When the alternate form is used for hex and binary, the `0x`/`0b` length is subtracted from the
-leading zeros.  This matches [`core::fmt` behavior](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=b11809759f975e266251f7968e542756). No further
-customization is supported (at the moment).
+## Alternate printing
```

Comment:
could you include in this section that `{}, "hello"` is formatted as `hello` but that `{:#}, "hello"` is formatted as `"hello"` and note that this matches `core::fmt` behavior

Reply:
I guess you mean `{:?}`? I am generally unsure what the `?` does anyways, since we don't have separate `Debug` formatting.

Reply:
I'll merge it without this being resolved and will raise the question in zulip :)